inbox/auth/oauth.py

```diff
@@ -90,29 +91,24 @@ def authenticate_imap_connection(self, account, conn):
             )
             raise
 
-    def _get_user_info(self, access_token):
+    def _get_user_info(self, session_dict):
+        access_token = session_dict["access_token"]
+        request = urllib.request.Request(
+            self.OAUTH_USER_INFO_URL,
+            headers={"Authorization": "Bearer {}".format(access_token)},
+        )
         try:
-            response = requests.get(
-                self.OAUTH_USER_INFO_URL, params={"access_token": access_token}
-            )
-        except requests.exceptions.ConnectionError as e:
+            response = urllib.request.urlopen(request)
+        except urllib.error.HTTPError as e:
+            if e.code == 401:
+                raise OAuthError()
+            log.error("user_info_fetch_failed", error_code=e.code, error=e)
+            raise ConnectionError()
+        except urllib.error.URLError as e:
             log.error("user_info_fetch_failed", error=e)
             raise ConnectionError()
 
-        userinfo_dict = response.json()
-
-        if "error" in userinfo_dict:
-            assert userinfo_dict["error"] == "invalid_token"
```

Comment:
That's an oddly specific condition, any idea what this was testing against?

Reply:
- The expected OpenID response would be a 401, like this: https://openid.net/specs/openid-connect-core-1_0.html#UserInfoError
- I found this doc in their YouTube API that suggests it returns a 400 with `invalid_token`. https://developers.google.com/youtube/v3/live/guides/auth/client-side-web-apps 

I can do a test to see what Google actually returns.

It also seems that the original code doesn't check the HTTP response code, which is why it looks the error response. I don't think the original code is particularly great.

Reply:
Yep agreed that the change is better, I was mostly just curious.